admin-dev/themes/new-theme/js/components/form/submittable-input.js

```diff
@@ -55,20 +55,21 @@ export default class SubmittableInput {
    */
   init() {
     const inputs = `${this.wrapperSelector} ${this.inputSelector}`;
+    const that = this;
 
     $(document).on('focus', inputs, (e) => {
       this.refreshButtonState(e.currentTarget, true);
     });
     $(document).on('input blur', inputs, (e) => {
       this.refreshButtonState(e.currentTarget);
     });
-    $(document).on('click', `${this.wrapperSelector} ${this.buttonSelector}`, (e) => {
-      this.submitInput(e);
+    $(document).on('click', `${this.wrapperSelector} ${this.buttonSelector}`, function () {
+      that.submitInput(this);
```

Comment:
I don't understand why you replaced this function by returning the input instead of the event? Why is it better?
But in this case, you should rename the parameter in `submitInput` parameter with `input` instead of `e` because it's a bit confusing

Reply:
The event sticks to the form, not to the input.

Previously, the one who developed this component was using event.target, and it was returning the form, so you were not able to play with the button around the input which was changing :/

Reply:
OK, and any reason why you replaced the arrow function? Which forces you to store `this` into `that`

Since I started using arrow function I always think it's more convenient because `this` is accessible (and correct) right away without relying on storing it or rebinding the function But maybe there's a reason why you chose not to use an arrow function here?

Reply:
Oh I get it! That's because you need the `this` from the click function to match the clicked input Although this could be done using `event.currentTarget` no?

Reply:
event.currentTarget return document, not the element you clicked on exactly